website/addons/citations/provider.py

```diff
@@ -138,14 +137,16 @@ def citation_list(self, node_addon, user, list_id, show='all'):
                 ]
 
             if show in ('all', 'citations'):
+                citations, next_page = node_addon.api.get_list(list_id, page)
                 contents += [
                     self.serializer(
                         node_settings=node_addon,
                         user_settings=user_settings,
                     ).serialize_citation(each)
-                    for each in node_addon.api.get_list(list_id)
+                    for each in citations
                 ]
-
+                if self.next_is_truthy(next_page):
```

Comment:
I second @sloria. If `next_page` would be `None` for Mendeley or `0` for Zotero (or otherwise truthy), then `if next_page` should work here